raiden/network/transport/matrix/rtc/web_rtc.py

```diff
@@ -682,41 +695,34 @@ def _on_ice_gathering_state_change(
         for candidate in rtc_ice_candidates:
             candidate = {
                 "candidate": f"candidate:{candidate_to_sdp(candidate)}",
-                "sdpMid": candidate.sdpMid if candidate.sdpMid is not None else SDP_MID_DEFAULT,
+                "sdpMid": candidate.sdpMid if candidate.sdpMid is not None else _SDP_MID_DEFAULT,
                 "sdpMLineIndex": candidate.sdpMLineIndex is not None
                 if candidate.sdpMLineIndex
-                else SDP_MLINE_INDEX_DEFAULT,
+                else _SDP_MLINE_INDEX_DEFAULT,
             }
             candidates.append(candidate)
 
-        wrap_callback(
-            callback=candidates_callback,
-            candidates=candidates,
-            partner_address=rtc_partner.partner_address,
-        )
+        wrap_callback(callback=candidates_callback, candidates=candidates)
 
 
-def _on_ice_connection_state_change(
-    rtc_partner: _RTCPartner, closed_callback: Callable[[Address], None]
-) -> None:
-    ice_connection_state = rtc_partner.peer_connection.iceConnectionState
-    rtc_partner.log.debug("Ice connection state changed", signaling_state=ice_connection_state)
+def _on_ice_connection_state_change(conn: _RTCConnection) -> None:
+    ice_connection_state = conn.peer_connection.iceConnectionState
+    conn.log.debug("ICE connection state changed", signaling_state=ice_connection_state)
 
-    if ice_connection_state in [ICEConnectionState.CLOSED.value, ICEConnectionState.FAILED.value]:
-        asyncio.create_task(rtc_partner.reset())
-        wrap_callback(callback=closed_callback, partner_address=rtc_partner.partner_address)
```

Comment:
Are you sure it is correct to remove the `closed_callback` here?

Reply:
Well, that callback did two things:
- slept 3s
- tried to establish the WebRTC channel again

My thinking was that was unnecessary. If the channel is closed, the other could still remain available. If no channels are available, then the channel creation will be triggered by health check anyway.

Reply:
In my opinion we definitely should have a retry mechanism in place. This would avoid slow payments due to channel failure. Imagine a channel goes down because of a connection issue. Then we should wait for the next healthcheck (which might occur upon the next payment) resulting in a slow payment. 

We should rather retry a fixed amount of times (lets say 3 times) if a channel does not get created, we can assume that the partner is offline and stop retrying, waiting for the next health check. 

This optimistic approach of retrying prevents issues for example when a peer roams to a different network or the network connection breaks (which happens once in a while)

Reply:
Changed to invoke health check that takes care of retrying 3 times.